std_execution.bs

```diff
@@ -3022,3 +3039,152 @@ are all well-formed.
+      using <i>single-sender-value-type</i> = <i>see below</i>;
+
+    // exposition only
+    template&lt;class S>
+      concept <i>single-typed-sender</i> =
+        typed_sender&lt;S> &&
+        requires { typename <i>single-sender-value-type</i>&lt;S>; };
+
+    template&lt;<i>single-typed-sender</i> Sender, class Promise>
+      <i>unspecified</i> as_awaitable(Sender&& s, Promise& p);
+    </pre>
+
+2. Alias template <i>single-sender-value-type</i> is defined as follows:
+
+    - If `sender_traits<remove_cvref_t<S>>::value_types<Tuple, Variant>` would have the form `Variant<Tuple<T>>`, then <code><i>single-sender-value-type</i>&lt;S></code> is an alias for type `T`.
+
+    - Otherwise, if `sender_traits<remove_cvref_t<S>>::value_types<Tuple, Variant>` would have the form `Variant<Tuple<>>` or `Variant<>`, then <code><i>single-sender-value-type</i>&lt;S></code> is an alias for type `void`.
+
+    - Otherwise, <code><i>single-sender-value-type</i>&lt;S</code> is ill-formed.
+
+3. For some expressions `s` and `p` such that `decltype((s))` is `S` and `decltype((p))` is `P&`, `as_awaitable(s, p)` returns <code><i>sender-awaitable</i>{s, p}</code>, where  <code><i>sender-awaitable</i></code> is an unspecified class type equivalant to the following:
+
+    <pre highlight="c++">
+    class <i>sender-awaitable</i> {
+      struct unit {}; // exposition only
+      using value_t = <i>single-sender-value-type</i>&lt;S>;
+      using result_t = conditional_t&lt;is_void_v&lt;value_t>, unit, value_t>;
+      struct <i>awaitable-receiver</i>; // exposition only
+
+      variant&lt;monostate, result_t, exception_ptr> result_{}; // exposition only
+      connect_result_t&lt;S, <i>awaitable-receiver</i>> state_;  // exposition only
+
+     public:
+      <i>sender-awaitable</i>(S&& s, P& p);
+      bool await_ready() const noexcept { return false; }
+      void await_suspend(coro::coroutine_handle&lt;>) noexcept { start(state_); }
+      value_t await_resume();
+    };
+    </pre>
+
+    1. <code><i>awaitable-receiver</i></code> is a receiver type equivalent to the following:
+
+        <pre highlight=c++>
+        struct <i>awaitable-receiver</i> { // exposition only
+          variant&lt;monostate, result_t, exception_ptr>* result_ptr_;
+          coroutine_handle&lt;P> continuation_;
+          // ... <i>see below</i>
+        };
+        </pre>
+
+        Let `r` be an rvalue expression of type <code><i>awaitable-receiver</i></code>, let `v` be an expression of type `result_t`, let `e` be an expression of type `exception_ptr`, let `p` be an lvalue reference to the coroutine promise type, let `c` be a customization point object, and let `as...` be a pack of arguments. Then:
+
+          - If `value_t` is `void`, then `execution::set_value(r)` is expression-equivalent to  `(result_ptr_->emplace<1>(v), continuation_.resume())`; otherwise, `execution::set_value(r, v)` is expression-equivalent to `(result_ptr_->emplace<1>(v), continuation_.resume())`.
```

Comment:
Lol, the "if" and the "otherwise" cases are the same. 😳

Reply:
I think you need to drop the `v` parameter to `emplace<1>(v)` in the "If `value_t` is `void`" case.

Reply:
Actually, no. If `value_t` is `void`, then `result_t` is `unit`, and you _do_ want to emplace a `unit` object in the variant.

Reply:
Yes, but `unit` is default constructible so `emplace<1>()` should be just fine.
Also, there is no definition of `v` for the `void` branch of this logic.